source/utilities.tex

```diff
@@ -12490,6 +12497,119 @@
 Nothing.
 \end{itemdescr}
 
+\indexlibrarymember{allocate_bytes}{polymorphic_allocator}%
+\begin{itemdecl}
+void* allocate_bytes(size_t nbytes, size_t alignment = alignof(max_align_t));
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to: \tcode{return memory_rsrc->allocate(nbytes, alignment);}
+
+\pnum
+\begin{note}
+The return type is \tcode{void*} (rather than, e.g., \tcode{byte*})
+to support conversion to an arbitrary pointer type \tcode{U*}
+by \tcode{static_cast<U*>}, thus facilitating construction of a \tcode{U}
+object in the allocated memory.
+\end{note}
+\end{itemdescr}
+
+\indexlibrarymember{deallocate_bytes}{polymorphic_allocator}%
+\begin{itemdecl}
+void deallocate_bytes(void* p, size_t nbytes, size_t alignment = alignof(max_align_t));
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to \tcode{memory_rsrc->deallocate(p, nbytes, alignment)}.
+\end{itemdescr}
+
+\indexlibrarymember{allocate_object}{polymorphic_allocator}%
+\begin{itemdecl}
+template <class T>
+  T* allocate_object(size_t n = 1);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Allocates memory suitable for holding
+an array of \tcode{n} objects of type \tcode{T}, as follows:
+\begin{itemize}
+\item
+  if \tcode{SIZE_MAX / sizeof(T) < n}, throws \tcode{length_error},
+\item
+  otherwise equivalent to:
+\begin{codeblock}
+return static_cast<T*>(allocate_bytes(n*sizeof(T), alignof(T)));
+\end{codeblock}
+\end{itemize}
+
+\pnum
+\begin{note}
+\tcode{T} is not deduced and must therefore be provided as a template argument.
+\end{note}
+\end{itemdescr}
+
+\indexlibrarymember{deallocate_object}{polymorphic_allocator}%
+\begin{itemdecl}
+template <class T>
+  void deallocate_object(T* p, size_t n = 1);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Equivalent to \tcode{deallocate_bytes(p, n*sizeof(T), alignof(T))}.
+\end{itemdescr}
+
+\indexlibrarymember{new_object}{polymorphic_allocator}%
+\begin{itemdecl}
+template <class T, class CtorArgs...>
+  T* new_object(CtorArgs&&... ctor_args);
+\end{itemdecl}
+
+\begin{itemdescr}
+\pnum
+\effects
+Allocates and constructs an object of type \tcode{T}, as follows:
+equivalent to:
```

Comment:
Can we do this instead to get rid of the double :s?
```
... as follows.\linebreak
Equivalent to:
```

Reply:
Agreed. Fixed.